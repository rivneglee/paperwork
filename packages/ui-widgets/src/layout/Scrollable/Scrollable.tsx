import React, { FunctionComponent } from 'react';

import ScrollBars from 'react-custom-scrollbars';

interface Props {
  className?: string;
  setRef?: (ref: any) => void;
}

const Scrollable: FunctionComponent<Props> = ({
 children, className = '', setRef,
}) => {
  const setParentRef = (ref: any) => {
    if (ref && setRef) {
      setRef(ref.firstChild.firstChild);
    }
  };

  return (
    <div className={`pw-scrollable ${className}`} ref={setParentRef}>
      <ScrollBars autoHide>
        {children}
      </ScrollBars>
    </div>
  );
};

export default Scrollable;
