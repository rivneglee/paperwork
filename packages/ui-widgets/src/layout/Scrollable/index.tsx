export { default as Scrollable } from './Scrollable';
