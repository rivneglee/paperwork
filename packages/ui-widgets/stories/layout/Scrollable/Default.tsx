import React from 'react';

import { Card, Scrollable, Icons, List } from '../../../src';

export default () => (
  <div style={{ background: '#ebeef1', padding: '2.4rem' }}>
    <Card header={
      <h3>Beautiful scroll bar</h3>
    }>
      <Scrollable>
        <List>
          {
            Array.from({ length: 10 }).map(() => (
              <>
                <List.Item onClick={close} icon={<Icons.Form/>}>
                  Forms
                </List.Item>
                <List.Item onClick={close} icon={<Icons.Chart/>}>
                  Reports
                </List.Item>
              </>
            ))
          }
        </List>
      </Scrollable>
    </Card>
  </div>
);
