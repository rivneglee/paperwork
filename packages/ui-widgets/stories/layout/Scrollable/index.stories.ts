import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('Scrollable', module)
  .add('default', Default);
