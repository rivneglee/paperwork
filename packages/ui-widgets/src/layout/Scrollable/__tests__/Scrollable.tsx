import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Scrollable from '../Scrollable';

describe('Scrollble', () => {
  const Content = () => <>'Content'</>;

  it('should render without custom css', () => {
    const wrapper: ShallowWrapper = shallow(
      <Scrollable>
        <Content />
      </Scrollable>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with custom css', () => {
    const wrapper: ShallowWrapper = shallow(
      <Scrollable className="custom-css">
        <Content />
      </Scrollable>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
