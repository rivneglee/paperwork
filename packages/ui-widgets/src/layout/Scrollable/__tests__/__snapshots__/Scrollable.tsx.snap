// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scrollble should render with custom css 1`] = `
<div
  className="pw-scrollable custom-css"
>
  <Scrollbars
    autoHeight={false}
    autoHeightMax={200}
    autoHeightMin={0}
    autoHide={true}
    autoHideDuration={200}
    autoHideTimeout={1000}
    hideTracksWhenNotNeeded={false}
    renderThumbHorizontal={[Function]}
    renderThumbVertical={[Function]}
    renderTrackHorizontal={[Function]}
    renderTrackVertical={[Function]}
    renderView={[Function]}
    tagName="div"
    thumbMinSize={30}
    universal={false}
  >
    <Content />
  </Scrollbars>
</div>
`;

exports[`Scrollble should render without custom css 1`] = `
<div
  className="pw-scrollable "
>
  <Scrollbars
    autoHeight={false}
    autoHeightMax={200}
    autoHeightMin={0}
    autoHide={true}
    autoHideDuration={200}
    autoHideTimeout={1000}
    hideTracksWhenNotNeeded={false}
    renderThumbHorizontal={[Function]}
    renderThumbVertical={[Function]}
    renderTrackHorizontal={[Function]}
    renderTrackVertical={[Function]}
    renderView={[Function]}
    tagName="div"
    thumbMinSize={30}
    universal={false}
  >
    <Content />
  </Scrollbars>
</div>
`;
